cachi2/core/package_managers/rubygems/parser.py

```diff
@@ -67,12 +70,62 @@ class PathDependency(GemMetadata):
 
 def parse_gemlock(package_dir: RootedPath) -> list[BundlerDependency]:
     """Parse a Gemfile.lock file and return a list of dependencies."""
-    return NotImplemented
+    gemlock_path = package_dir.join_within_root(GEMFILE_LOCK)
+    if not gemlock_path.path.exists():
+        reason = f"Gemfile.lock not found at {gemlock_path.path}"
+        solution = "Please ensure the Gemfile.lock file is present in the package directory"
```

Comment:
FWIW you'd call the Ruby lockfile parsing script somewhere around here, but that would change as more repo validations would be done first elsewhere.